### DISCUSSION TITLE
Quire PDF for Print-on-Demand

### DISCUSSION THREAD

**Discussion by chrisdaaz** [thegetty/quire#46] [Get Help] [answered]
I'm trying to generate an interior PDF file for a print-on-demand service ([Lulu](https://www.lulu.com/)). Do you have recommendations on how to interact with PrinceXML PDF settings? I need to be able to reset the page size to 6"x9" and possible adjust the margins. I'm not sure if there are CSS styles I should add to `static/custom.css` or if I need to use PrinceXML directly or...

Any advice?

**Comment by geealbers** [comment, marked as answer]
Hi @chrisdaaz. In the [`source/css/variable.scss`](https://github.com/thegetty/quire/blob/24113d3bbe96abd99afdd2c05274d9b1c13d282d/themes/default/source/css/variables.scss#L58-L71) file of your project's theme, there are a number of print/pdf-related variables exposed there including for the page width and height. 

```scss
// Print/PDF stylesheet
// -----------------------------------------------------------------------------
$print-width: 8.5in;
$print-height: 11in;
$print-bleed: .125in;

$print-base-font-size: 8.5pt;
$print-text-color: $black;

$print-splash-color: $off-white;

$print-entry-image-color: $rich-black;
$print-entry-caption-color: $white;
$print-entry-image-display: all; // first | all
```

You'll find even more at the top of the [`source/css/print.scss`](https://github.com/thegetty/quire/blob/main/themes/default/source/css/print.scss) file. Everything you need for adjusting the file to your liking for print-on-demand file should be available through those variables.

**There are though a couple gothcas here that we still need to fix, and that you should be aware of:**

First, in `source/css/print.scss` we define both a `$print-base-text-column-width` as well as inner and outer margins. The problem with this is that if you add up the values of the two margins and the text column and they don’t equal the value you have set as a print-width, you can get some unexpected results.

Second, we have some well-intentioned but ultimately flawed logic built in that can also lead to unexpected results as some margins are automatically adjusted if the print width is less than or equal to 7" and again if it’s less than or equal to 10".

```scss
@if $print-width <= 7 {
  $print-inner-margin: $print-width - $print-base-text-column-width - $print-outer-margin;
}

@if $print-height <= 10 {
  $print-top-margin: 0.5in;
}
```

**Reply by chrisdaaz** [reply]
Thank you! This is exactly was I was looking for!
